modules/backend/calendar/WorkUnit.cpp

```diff
@@ -0,0 +1,22 @@
+#include "WorkUnit.hpp"
+
+namespace calendar
+{
+
+WorkUnit::WorkUnit(clinic::staff::doctor::iDoctor* doc, clinic::room::iRoom* room) : doc_ptr_(doc), room_ptr_(room)
+{
+}
+void WorkUnit::addNewTimeSlot(std::unique_ptr<calendar::TimeSlot> timeSlot)
+{
+    _slot_collection.push_back(std::move(timeSlot));
```

Comment:
suggestion to use `emplace_back` as we might be calling createTimeSlot(...) and moving it straight to WorkUnit object

Reply:
done